clients/client/src/main/java/org/projectnessie/client/api/CreateReferenceBuilder.java

```diff
@@ -29,10 +29,23 @@
  * @since {@link NessieApiV1}
  */
 public interface CreateReferenceBuilder {
+  /**
+   * Configure source reference name.
+   *
+   * @param sourceRefName is the name of the reference which contains the hash to be used for the
+   *     new reference. Default reference name will be used if not configured.
+   */
   CreateReferenceBuilder sourceRefName(
       @Pattern(regexp = Validation.REF_NAME_REGEX, message = Validation.REF_NAME_MESSAGE)
           String sourceRefName);
 
+  /**
+   * Configure new reference name and hash to be used.
+   *
+   * @param reference is the {@link Reference} object with the new reference name and hash from
+   *     source reference. If the hash is null, it will be set to reference the beginning of time
+   *     (aka no_ancestor hash).
```

Comment:
`A specific sub-class of Reference (Branch of Tag) defining the name and hash for the new reference to be created. The hash must be reachable from the {@link #sourceRefName source reference}`?